### PR TITLE
Deploy module

### PR DESCRIPTION
**Pull request by IonesioJunior**
Create a new module to deploy grid components on different platforms.
 - Add BaseDeployment abstract class.
 - Add NodeDeployment class.
 - Update ["Launch Websocket Node on Heroku"](https://github.com/OpenMined/Grid/blob/deploy_module/examples/experimental/Launch_Websocket_Grid_worker.ipynb) notebook.

### CODE REVIEW

File: grid/deploy/component.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            commands: list of commands used in deploy process
+        """
+        self.env_vars = env_vars
+        self.verbose = verbose
+        self.logs = list()
+        self.commands = list()
+
+    def deploy_on_heroku(self):
+        """ Method used to deploy component on heroku platform. """
+        raise RuntimeError("Heroku deployment not specified!")
```
Comment (mari-linhares) [inline grid/deploy/component.py]: Raise NotImplementedError instead.

File: grid/deploy/component.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    def deploy_on_container(self):
+        """ Method used to deploy component using docker containers. """
+        raise RuntimeError("Container deployment not specified!")
```
Comment (mari-linhares) [inline grid/deploy/component.py]: same

File: grid/deploy/component.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+class BaseDeployment(ABC):
+    """ Abstract Class used to instantiate generic attributes for other deployment classes. """
+
+    def __init__(self, env_vars, verbose: bool = True):
+        """
+        Args:
+            env_vars: Environment vars used to configure component
+            verbose: Used to define level of verbosity
+            logs: list of logs used in deploy process
```
Comment (mari-linhares) [inline grid/deploy/component.py]: list -> List

File: grid/deploy/component.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            commands: list of commands used in deploy process
```
Comment (mari-linhares) [inline grid/deploy/component.py]: list -> List

File: grid/deploy/component.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        raise RuntimeError("Container deployment not specified!")
+
+    def _check_dependency(
+        self,
+        lib="git",
+        check="usage:",
+        error_msg="Error: please install git.",
+        verbose=False,
+    ):
+        """ This method checks if the environment have a specific dependency.
```
Comment (mari-linhares) [inline grid/deploy/component.py]: have -> has

File: grid/deploy/component.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            Args:
+                dependency_lib : libs that will be verified.
+                check: specific string to check if app was installed.
```
Comment (mari-linhares) [inline grid/deploy/component.py]: specific -> Specific

File: grid/deploy/component.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    def _check_dependency(
+        self,
+        lib="git",
+        check="usage:",
+        error_msg="Error: please install git.",
+        verbose=False,
+    ):
+        """ This method checks if the environment have a specific dependency.
+            Args:
+                dependency_lib : libs that will be verified.
```
Comment (mari-linhares) [inline grid/deploy/component.py]: libs -> Libs

File: grid/deploy/component.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+                check: specific string to check if app was installed.
+                error_msg: If not installed, raise an Exception with this.
+                verbose: Used to define level of verbosity.
+            Raises:
+                Exception: if not installed, raise a standard Exception
```
Comment (mari-linhares) [inline grid/deploy/component.py]: Could we raise a more specific exception? Maybe a runtime exception is better.

File: grid/deploy/component.py [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+        """
+        if verbose:
+            sys.stdout.write("\tChecking for " + str(lib) + " dependency...")
+        o = gr_utils.exec_os_cmd(lib)
```
Comment (mari-linhares) [inline grid/deploy/component.py]: user a better name

File: grid/deploy/component.py [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+        outputs = list()
+
+        cmd = ""
+        for i in range(len(commands)):
+
+            if verbose:
+                print(logs[i] + "...")
+
+            cmd = "cd " + str(tmp_dir) + "; " + commands[i] + "; cd ..;"
+            o = gr_utils.exec_os_cmd(cmd)
```
Comment (mari-linhares) [inline grid/deploy/component.py]: use a better name

File: grid/deploy/node.py [hunk 11]
```diff
@@ -0,0 +1,184 @@
+from grid.deploy import BaseDeployment
+from grid import utils as gr_utils
+
+import sys
+import os
+
+
+class NodeDeployment(BaseDeployment):
```
Comment (mari-linhares) [inline grid/deploy/node.py]: I think this is not (and should not be) a general node deployment abstraction. It is actually a "HerokuNodeDeployment", right?

File: grid/deploy/node.py [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    """ An abstraction of grid node deployment process, the purpose of this class is set all configuration needed to deploy grid node application in different platforms."""
```
Comment (mari-linhares) [inline grid/deploy/node.py]: Update description

Comment (IonesioJunior) [reply grid/deploy/node.py]: Done! (b2843ea)

File: grid/deploy/node.py (hunk 11 above)
Comment (IonesioJunior) [reply grid/deploy/node.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 10 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 9 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 8 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 6 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 5 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 4 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 3 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 2 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/deploy/component.py (hunk 1 above)
Comment (IonesioJunior) [reply grid/deploy/component.py]: Done! (b2843ea)

File: grid/__init__.py [hunk 13]
```diff
@@ -41,7 +42,7 @@ def check_dependency(
 ):
     if verbose:
         sys.stdout.write("\tChecking for " + str(lib) + " dependency...")
-    o = gr_utils.exec_os_cmd(lib)
+    o = gr_utils.execute_command(lib)
```
Comment (mari-linhares) [inline grid/__init__.py]: Oh, I meant a better name to the `o` variable. Can you rename it to something like: `commands`. I should have been more explicit sorry.

File: grid/deploy/component.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+                dependency_lib : Libs that will be verified.
+                check: Specific string to check if app was installed.
+                error_msg: If not installed, raise an Exception with this.
+                verbose: Used to define level of verbosity.
+            Raises:
+                RuntimeError: If not installed, raise a RuntimeError Exception.
+        """
+        if verbose:
+            sys.stdout.write("\tChecking for " + str(lib) + " dependency...")
+        o = gr_utils.execute_command(lib)
```
Comment (mari-linhares) [inline grid/deploy/component.py]: same

File: grid/deploy/component.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+        outputs = list()
+
+        cmd = ""
+        for i in range(len(commands)):
+
+            if verbose:
+                print(logs[i] + "...")
+
+            cmd = "cd " + str(tmp_dir) + "; " + commands[i] + "; cd ..;"
+            o = gr_utils.execute_command(cmd)
```
Comment (mari-linhares) [inline grid/deploy/component.py]: same

File: grid/deploy/component.py [hunk 16]
```diff
@@ -0,0 +1,102 @@
+import sys
```
Comment (mari-linhares) [inline grid/deploy/component.py]: Why is this file named component? maybe rename it to `base_deployment.py`

**Pull request merged**